### PR TITLE
Upgrade pnpm to `v9.0.1` 

### PR DESCRIPTION
**Pull request by ProchaLu**
This PR upgrades pnpm to `v9.0.1`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ COPY ./docker/package.json ./docker/pnpm-lock.yaml ./
 RUN apk update
 RUN apk add --no-cache git postgresql python3 py3-pip build-base bash
 
-RUN corepack enable && corepack prepare pnpm@9.0.0-beta.2 --activate
+RUN corepack enable && corepack prepare pnpm@9.0.1 --activate
```
Comment (karlhorky) [inline Dockerfile:14]: did we have exact versions before?

File: package.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
   },
   "engines": {
     "node": ">=18",
-    "pnpm": ">=9.0.0-beta.2"
+    "pnpm": ">=9.0.1"
```
Comment (karlhorky) [inline package.json:82]: did we have a version before?

Comment (ProchaLu) [reply package.json]: yes, it was added in this [commit](https://github.com/upleveled/preflight/commit/48967763de2c0cc66049377833dad5ae01b1b57f) with `v9.0.0-beta.3`, there was no `pnpm` version in the engine section. I removed it again 
[8b91f0c](https://github.com/upleveled/preflight/pull/525/commits/8b91f0cc9596eca7faae6a58efb97fd860640f52)

**Review by karlhorky — APPROVED** [review]

**Comment by karlhorky** [comment]
@ProchaLu this broke Docker builds:

```
#11 3.840 Preparing pnpm@latest for immediate activation...
#11 DONE 4.4s

#12 [ 7/10] RUN pnpm install --frozen-lockfile
#12 1.034  ERR_PNPM_LOCKFILE_BREAKING_CHANGE  Lockfile /preflight/pnpm-lock.yaml not compatible with current pnpm
#12 1.034 
#12 1.034 Run with the --force parameter to recreate the lockfile.
#12 ERROR: process "/bin/sh -c pnpm install --frozen-lockfile" did not complete successfully: exit code: 1
------
 > [ 7/10] RUN pnpm install --frozen-lockfile:
1.034  ERR_PNPM_LOCKFILE_BREAKING_CHANGE  Lockfile /preflight/pnpm-lock.yaml not compatible with current pnpm
1.034 
1.034 Run with the --force parameter to recreate the lockfile.
```

https://github.com/upleveled/preflight/actions/runs/8721156129/job/23924246682

Can you investigate and fix? (if it takes longer than today, then maybe open an issue to track it).

May be related to the old v7 lockfile in `docker/pnpm-lock.yaml`:

https://github.com/upleveled/preflight/blob/594cd50b1091f61522de66d1ac0df584c18f36a8/docker/pnpm-lock.yaml#L1

**Pull request merged**